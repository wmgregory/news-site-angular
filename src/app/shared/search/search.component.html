<div id="search-component">

  <div class="form-inline">
    <div class="form-group mb-2">
      <h4>News Search</h4>
    </div>

    <div class="form-group mx-sm-3 mb-2">
      <label for="search" class="sr-only">Password</label>
      <input class="form-control" type="text" id="search"
             #search
             [(ngModel)]="keyword"
             (keyup.enter)="addKeyword()"
             placeholder="type here...." />
    </div>

    <button class="btn btn-primary mb-2" type="button" (click)="addKeyword()">search</button>
    <button class="btn btn-primary mb-2" type="button" (click)="clear()">clear</button>
  </div>

  <h5>Keywords</h5>
  <ul class="search-keywords">
    <li *ngFor="let keyword of keywords">
      <button type="button" class="btn btn-light" (click)="removeKeyword(keyword)">{{ keyword }} x</button></li>
  </ul>
</div>
