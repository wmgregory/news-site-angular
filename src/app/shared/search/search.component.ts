import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';

@Component({
  selector: 'app-search',
  templateUrl: './search.component.html',
  styleUrls: ['./search.component.scss']
})
export class SearchComponent implements OnInit {

  @Output() search: EventEmitter<any> = new EventEmitter();

  public keyword: string = '';
  public keywords: string[] = [];

  constructor() {}

  ngOnInit() {}

  addKeyword(): void {
    const trimmed = this.keyword.trim();

    console.log(trimmed);

    if (trimmed) {
      this.keywords.push(trimmed);

      this.search.emit(this.keywords.toString());
      this.keyword = '';
    }
  }

  removeKeyword(keyword: string): void {
    this.keywords = this.keywords.filter((item) => item !== keyword);
    this.search.emit(this.keywords.toString());
  }

  clear(): void {
    this.keywords = [];
    this.search.emit('');
    this.keyword = '';
  }
}
