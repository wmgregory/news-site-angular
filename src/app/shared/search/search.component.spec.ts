import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { FormsModule } from '@angular/forms';

import { SearchComponent } from './search.component';

describe('SearchComponent', () => {
  let component: SearchComponent;
  let fixture: ComponentFixture<SearchComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ SearchComponent ],
      imports: [ FormsModule ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SearchComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  describe('keywords', () => {

    it('should add a keyword', () => {
      component.keyword = 'updated value';
      component.addKeyword();

      fixture.detectChanges();

      expect(component.keywords).toEqual(['updated value']);
    });

    it('should remove a keyword', () => {
      component.keywords = ['remove'];

      fixture.detectChanges();

      expect(component.keywords).toEqual(['remove']);

      component.removeKeyword('remove');

      fixture.detectChanges();

      expect(component.keywords).toEqual([]);
    });

    it('should clear all keywords', () => {
      component.keywords = ['remove', 'all', 'words'];

      fixture.detectChanges();

      expect(component.keywords).toEqual(['remove', 'all', 'words']);

      component.clear();

      fixture.detectChanges();

      expect(component.keywords).toEqual([]);
    });

  });
});
