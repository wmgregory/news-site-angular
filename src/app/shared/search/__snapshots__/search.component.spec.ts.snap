// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent should create 1`] = `
<app-search
  keyword=""
  keywords={[Function Array]}
  search={[Function EventEmitter]}
>
  <div
    id="search-component"
  >
    
  
    
    <div
      class="form-inline"
    >
      
      
      <div
        class="form-group mb-2"
      >
        
        
        <h4>
          News Search
        </h4>
        
      
      </div>
      
  
      
      <div
        class="form-group mx-sm-3 mb-2"
      >
        
        
        <label
          class="sr-only"
          for="search"
        >
          Password
        </label>
        
        
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          id="search"
          ng-reflect-model=""
          placeholder="type here...."
          type="text"
        />
        
      
      </div>
      
  
      
      <button
        class="btn btn-primary mb-2"
        id="addKeyword"
        type="button"
      >
        search
      </button>
      
      
      <button
        class="btn btn-primary mb-2"
        id="clear"
        type="button"
      >
        clear
      </button>
      
    
    </div>
    
  
    
    <h5>
      Keywords
    </h5>
    
    
    <ul
      class="search-keywords"
    >
      
      
      
      
    
    </ul>
    
  
  </div>
  
</app-search>
`;
